### PR TITLE
Expands logging to CopyFileLinksService

### PR DESCRIPTION
**Pull request by mereghost**
## Related WP: [OP#56357](https://community.openproject.org/projects/openproject/work_packages/56357)

Just more of the same, just adding some extra logging here and there and some cleanups

### CODE REVIEW

File: modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/files_info_query.rb [hunk 1]
```diff
@@ -33,6 +33,7 @@ module Peripherals
     module StorageInteraction
       module OneDrive
         class FilesInfoQuery
+          include TaggedLogging
```
Comment (Kharonus) [inline modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/files_info_query.rb:36]: ❓ isn't there a usage of `with_tagged_logger` needed and missing in the one drive query?

File: modules/storages/app/services/storages/file_links/copy_file_links_service.rb [hunk 2]
```diff
@@ -122,13 +127,14 @@ def source_files_info(source_file_links)
 
       def target_files_map
         Peripherals::Registry
-          .resolve("#{@source.storage.short_provider_type}.queries.file_path_to_id_map")
-          .call(storage: @source.storage, auth_strategy:, folder: Peripherals::ParentFolder.new(@target.project_folder_location))
+          .resolve("#{@target.storage.short_provider_type}.queries.file_path_to_id_map")
+          .call(storage: @target.storage, auth_strategy:, folder: Peripherals::ParentFolder.new(@target.project_folder_location))
```
Comment (Kharonus) [inline modules/storages/app/services/storages/file_links/copy_file_links_service.rb:131]: 🟢 uuh, was that a big bug? :D what were possible consequences? 😂

File: modules/storages/spec/services/storages/file_links/copy_file_links_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        expect(link.origin_id).to eq(source_links[index].origin_id)
+      end
+    end
+  end
+
+  context "when AMPF is enabled" do
+    let(:files_info) { class_double(Storages::Peripherals::StorageInteraction::Nextcloud::FilesInfoQuery) }
+    let(:file_path_to_id) { class_double(Storages::Peripherals::StorageInteraction::Nextcloud::FilePathToIdMapQuery) }
+    let(:auth_strategy) do
+      Storages::Peripherals::StorageInteraction::AuthenticationStrategies::Strategy.new(key: :basic_auth)
```
Comment (Kharonus) [inline modules/storages/spec/services/storages/file_links/copy_file_links_service_spec.rb:44]: 🟡 why not the "usual" usage as `Storages::Peripherals::StorageInteraction::AuthenticationStrategies::BasicAuth.strategy`?

**Review by Kharonus — CHANGES_REQUESTED** [review]
I only have two small issues. If there is a good reason for it, speak up! :P

File: modules/storages/app/services/storages/file_links/copy_file_links_service.rb (hunk 2 above)
Comment (mereghost) [reply modules/storages/app/services/storages/file_links/copy_file_links_service.rb:131]: Not many, as we didn't expect 2 different instances, but... now it is handled.

In other words: the target project_storage is a copy of the source, so it should not  be a different storage at all but it is clearer this way.

File: modules/storages/spec/services/storages/file_links/copy_file_links_service_spec.rb (hunk 3 above)
Comment (mereghost) [reply modules/storages/spec/services/storages/file_links/copy_file_links_service_spec.rb:44]: I want the actual strategy for matching the expectations. Using the `.strategy` will create a new object every time and won't match later on, as it isn't a value object.

File: modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/files_info_query.rb (hunk 1 above)
Comment (mereghost) [reply modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/files_info_query.rb:36]: Good catch

**Review by Kharonus — APPROVED** [review]
LGTM

**Pull request merged**